okhttp/src/main/java/com/squareup/okhttp/Headers.java

```diff
@@ -130,6 +131,20 @@ public Builder newBuilder() {
     return result.toString();
   }
 
+  public Map<String, List<String>> toMultimap() {
+    HashMap<String, List<String>> result = new HashMap<String, List<String>>();
```

Comment:
As a general rule we always use `LinkedHashMap`, and in this case you'll get predictable iteration order which corresponds to the first occurrence of each header.